test/elixir_sense/core/metadata_builder_test.exs

```diff
@@ -889,55 +935,19 @@ defmodule ElixirSense.Core.MetadataBuilderTest do
       |> string_to_state
 
     assert [
-             %VarInfo{name: :par1, positions: [{3, 20}], scope_id: 2},
-             %VarInfo{name: :par2, positions: [{3, 33}], scope_id: 2},
-             %VarInfo{name: :par3, positions: [{3, 39}], scope_id: 2},
-             %VarInfo{name: :par4, positions: [{3, 51}], scope_id: 2},
-             %VarInfo{name: :var_in1, positions: [{4, 5}], scope_id: 3},
-             %VarInfo{name: :var_in2, positions: [{5, 5}], scope_id: 3}
+             %VarInfo{name: :par1, positions: [{3, 20}], scope_id: 3},
+             %VarInfo{name: :par2, positions: [{3, 33}], scope_id: 3},
+             %VarInfo{name: :par3, positions: [{3, 39}], scope_id: 3},
+             %VarInfo{name: :par4, positions: [{3, 51}], scope_id: 3},
+             %VarInfo{name: :var_in1, positions: [{4, 5}], scope_id: 4},
+             %VarInfo{name: :var_in2, positions: [{5, 5}], scope_id: 4}
            ] = state |> get_line_vars(6)
 
     assert [
-             %VarInfo{name: :arg, positions: [{8, 14}], scope_id: 2}
+             %VarInfo{name: :arg, positions: [{8, 14}, {8, 24}], scope_id: 5}
            ] = state |> get_line_vars(8)
   end
 
-  test "guards do not define vars" do
```

Comment:
I added that test in https://github.com/elixir-lsp/elixir_sense/commit/1037edf659b401be872034285318cf7fd0fe9508 but looking at it now it was a workaround instead of a fix